### ISSUE TITLE
b_pie does not work in project default_options in subproject

### ISSUE THREAD

**Issue by pobrn** [mesonbuild/meson#12955]
**Describe the bug**

https://gitlab.freedesktop.org/pipewire/pipewire/-/jobs/56049278

```meson
project(..., default_options: ['b_pie=...'])
```
fails with an error in subprojects:
```
The Meson build system
Version: 1.4.0.rc1
Source dir: /tmp/asd
Build dir: /tmp/asd/build
Build type: native build
Project name: A
Project version: undefined
C compiler for the host machine: ccache cc (gcc 13.2.1 "cc (GCC) 13.2.1 20230801")
C linker for the host machine: cc ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64

Executing subproject B 


subprojects/B/meson.build:1:0: ERROR: In subproject B: Unknown options: "B:b_pie"

A full log can be found at /tmp/asd/build/meson-logs/meson-log.txt
```

**To Reproduce**

```meson
# ./meson.build
project('A', ['c'])
subproject('B')
```

```meson
# ./subprojects/B/meson.build
project('B', ['c'], default_options: ['b_pie=true'])
```

**Expected behavior**

No error.

---

Bisection: cc4cfbcad92945a1629c80664e1eb755c68905dd (#12922)
```
git bisect start
# status: waiting for both good and bad commits
# bad: [0cd74c96f1b58fe89ef9565ab40905d04538d7ed] cuda: respect host compiler `-Werror`
git bisect bad 0cd74c96f1b58fe89ef9565ab40905d04538d7ed
# status: waiting for good commit(s), bad commit known
# good: [614d436232d3a86518164cbe2b8af12db3bde009] Bump versions to 1.3.2 for release
git bisect good 614d436232d3a86518164cbe2b8af12db3bde009
# good: [7368795d13081d4928a9ba04d48498ca2442624b] Generate release notes for 1.3.0.
git bisect good 7368795d13081d4928a9ba04d48498ca2442624b
# good: [708a108cded4b25928b838483a95aad764622716] python module: correct the embedded path for bytecompiled files
git bisect good 708a108cded4b25928b838483a95aad764622716
# good: [6cd2ba42fff771d958c535f6024d2887823c0d9f] env2mfile: move language args to `built-in options`
git bisect good 6cd2ba42fff771d958c535f6024d2887823c0d9f
# good: [6cce6c51bf2ae4430d40331717bba8963ac475d8] cargo: Builder: Add support for "+", "+=", "if" and "foreach"
git bisect good 6cce6c51bf2ae4430d40331717bba8963ac475d8
# good: [937d1c639a9a83e0671e7e4cd3686ca3e8c9566b] nvcc: avoid adding `-Wpedantic` to compile lines
git bisect good 937d1c639a9a83e0671e7e4cd3686ca3e8c9566b
# bad: [54996132af6d0feaef6afd943c94ff25911ca67b] Merge pull request #12804 from joukewitteveen/dist-rewriter
git bisect bad 54996132af6d0feaef6afd943c94ff25911ca67b
# good: [76f6874e4841954a63ac9fb79deec353f50648d1] Fix base and compiler options not reconfigurable.
git bisect good 76f6874e4841954a63ac9fb79deec353f50648d1
# good: [524a66116b38a65210b7516781d26ed6efb690b9] Add MESONREWRITE to `meson dist` scripts
git bisect good 524a66116b38a65210b7516781d26ed6efb690b9
# bad: [cc4cfbcad92945a1629c80664e1eb755c68905dd] Fix unknown base options not detected in commandline arguments
git bisect bad cc4cfbcad92945a1629c80664e1eb755c68905dd
# first bad commit: [cc4cfbcad92945a1629c80664e1eb755c68905dd] Fix unknown base options not detected in commandline arguments
```

**Comment by dcbaker** [comment]
Sounds like: https://github.com/mesonbuild/meson/pull/12940. If you can verify that fixes your issue I can merge that today

**Comment by dcbaker** [comment]
Awesome. I merged that so the next 1.4 release will have that fix. Thanks!
